gitea_client.js

```diff
@@ -0,0 +1,165 @@
+ * @param normalizedUser Username
+ * @returns {*}
+ */
+const getUserIDByName = normalizedUser => {
+  const url = `${GITEA_HOST}/users/${normalizedUser}`;
+  return axios({
+    url,
+    headers: DEFAULT_HEADERS,
+    method: 'get'
+  });
+};
+
+/**
+ * Get gitea repository by name and user
+ * @param repoName Repository name
+ * @param normalizedUser Username
+ * @returns {*}
+ */
+const getRepoByName = ({ repoName }, normalizedUser) => {
+  const url = `${GITEA_HOST}/repos/${normalizedUser}/${repoName}`;
+  return axios({
+    url,
+    headers: DEFAULT_HEADERS,
+    method: 'get'
+  });
+};
+
+/**
+ * Clones an externally hosted repository (e.g. Github) into a new repository on
+ * gitea. A repository with that name must not exist in gitea under the specified
+ * user.
+ * @param repoName Repository name
+ * @param cloneUrl URL to the repository to clone from
+ * @param normalizedUser Username
+ * @returns {Promise<any>}
+ */
+const cloneExternalRepo = ({ repoName, cloneUrl }, normalizedUser) => {
+  const url = `${GITEA_HOST}/repos/migrate`;
+
+  return new Promise((resolve, reject) => {
+    getUserIDByName(normalizedUser)
+      .then(({ data }) => {
+        const payload = {
+          clone_addr: cloneUrl,
+          uid: data.id,
+          repo_name: repoName,
+          mirror: false
+        };
+        axios({
+          url,
+          method: 'post',
+          headers: DEFAULT_HEADERS,
+          data: payload
+        })
+          .then(resolve)
+          .catch(reject);
+      })
+      .catch(err => {
+        if (err.response && err.response.status === 422) {
+          console.log(`Skip creating repository ${repoName} because it already exists`);
+          return resolve();
+        }
+        return reject(err);
+      });
+  });
+};
+
+/**
+ * Creates a new empty repository on gitea with the specified name
+ * @param repoName Repository name
+ * @param normalizedUser Username
+ * @returns {Promise<any>}
+ */
+const createNewRepo = ({ repoName }, normalizedUser) => {
+  const url = `${GITEA_HOST}/admin/users/${normalizedUser}/repos`;
+  const data = {
+    name: repoName
+  };
+
+  return new Promise((resolve, reject) => {
+    axios({
+      method: 'post',
+      url,
+      data,
+      headers: DEFAULT_HEADERS
+    })
+      .then(resolve)
+      .catch(reject);
+  });
+};
+
+/**
+ * Create a repository for the given user. The repository argument is expected
+ * to be an object with the propertiey `repoName` and `cloneUrl`. If `cloneUrl`
+ * is missing, an empty repository will be created. Otherwise the repository
+ * contents are cloned from the specified clone url.
+ * @param username Username
+ * @param repo Repository name
+ * @returns {Promise<any>}
+ */
+exports.createRepoForUser = (username, repo) => {
+  const normalizedUser = normalizeUser(username);
+
+  return new Promise((resolve, reject) => {
+    if (!GITEA_HOST || !GITEA_TOKEN) {
+      return reject(new Error('Gitea not installed'));
```

Comment:
I tried running without setting these env vars, and I got this error in the logs as expected.
However, in the webapp UI it continues past the loading screen to the start of the walkthrough.
It would be better to fail at the loading screen with a useful error.

@aidenkeating Is that something the loading screen supports?

Reply:
Makes sense, i will change this to stop at the loading screen.